test/framework/src/main/java/org/elasticsearch/test/transport/CapturingTransport.java

```diff
@@ -84,6 +91,45 @@ public CapturedRequest(DiscoveryNode node, long requestId, String action, Transp
     private ConcurrentMap<Long, Tuple<DiscoveryNode, String>> requests = new ConcurrentHashMap<>();
     private BlockingQueue<CapturedRequest> capturedRequests = ConcurrentCollections.newBlockingQueue();
 
+    public TransportService createCapturingTransportService(Settings settings, ThreadPool threadPool, TransportInterceptor interceptor,
+                                                            Function<BoundTransportAddress, DiscoveryNode> localNodeFactory,
+                                                            @Nullable ClusterSettings clusterSettings, Set<String> taskHeaders) {
+        StubbableConnectionManager connectionManager = new StubbableConnectionManager(new ConnectionManager(settings, this, threadPool),
+            settings, this, threadPool);
+        connectionManager.setDefaultNodeConnectedBehavior((cm, discoveryNode) -> true);
+        connectionManager.setDefaultConnectBehavior((cm, discoveryNode) -> new Connection() {
```

Comment:
Bit late to the party here, but could this have been the following?

    connectionManager.setDefaultConnectBehavior((cm, discoveryNode) -> openConnection(discoveryNode, null));

That'd let me continue to override the default connection behaviour by subclassing `CapturingTransport`.

Reply:
Yeah I think that would work. Feel free to make that change in a PR where you need to override the method.

Reply:
Great. I opened https://github.com/elastic/elasticsearch/pull/33012.